core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -1360,45 +1360,62 @@ enum EventHandlingOutcome {
 
 /// Special handling for patch markers, when handling command events as in
 /// [WorkflowMachines::handle_command_event]
-fn change_marker_handling(
+fn patch_marker_handling(
     event: &HistoryEvent,
     mach: &Machines,
     next_event: Option<&HistoryEvent>,
 ) -> Result<EventHandlingOutcome> {
-    if !mach.matches_event(event) {
-        // Version markers can be skipped in the event they are deprecated
-        if let Some((patch_name, deprecated)) = event.get_patch_marker_details() {
+    let patch_machine = match mach {
+        Machines::PatchMachine(pm) => Some(pm),
+        _ => None,
+    };
+    let patch_details = event.get_patch_marker_details();
+    fn skip_one_or_two_events(next_event: Option<&HistoryEvent>) -> Result<EventHandlingOutcome> {
+        // Also ignore the subsequent upsert event if present
```

Comment:
Not introduced in this PR, but...
Are there cases where upsert search attributes follows a patch but the patch machine is not the one that issued that command?

I _think_ it's only possible if a patch was emitted before we started emitting SAs on patches but still worth supporting for correctness.

Reply:
It's probably technically possible, but, this check looks to see for the specific attribute key for versioning. So this could only happen if the user explicitly did that themselves somehow, which: A) would be super unlikely and B) means I have no way to differentiate anyways so nothing can be done.